From: Chris Lamb <lamby@debian.org>
Date: Mon, 16 May 2016 10:23:39 +0100
Subject: Skip logging tests as not all architectures support it yet

---
 tests/test_helper.tcl | 1 -
 1 file changed, 1 deletion(-)

Index: dotdeb-redis/tests/test_helper.tcl
===================================================================
--- dotdeb-redis.orig/tests/test_helper.tcl
+++ dotdeb-redis/tests/test_helper.tcl
@@ -40,7 +40,6 @@ set ::all_tests {
     integration/aof
     integration/rdb
     integration/convert-zipmap-hash-on-load
-    integration/logging
     unit/pubsub
     unit/slowlog
     unit/scripting
