From: Chris Lamb <lamby@debian.org>
Date: Fri, 30 Oct 2015 10:53:42 +0000
Subject: reproducible-build

---
 src/mkreleasehdr.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: dotdeb-redis/src/mkreleasehdr.sh
===================================================================
--- dotdeb-redis.orig/src/mkreleasehdr.sh
+++ dotdeb-redis/src/mkreleasehdr.sh
@@ -1,7 +1,7 @@
 #!/bin/sh
 GIT_SHA1=`(git show-ref --head --hash=8 2> /dev/null || echo 00000000) | head -n1`
 GIT_DIRTY=`git diff --no-ext-diff 2> /dev/null | wc -l`
-BUILD_ID=`uname -n`"-"`date +%s`
+BUILD_ID="Debian: `dpkg-parsechangelog -l../debian/changelog | sed -ne 's#Date: ##p'`"
 test -f release.h || touch release.h
 (cat release.h | grep SHA1 | grep $GIT_SHA1) && \
 (cat release.h | grep DIRTY | grep $GIT_DIRTY) && exit 0 # Already up-to-date
